text/0000-ssr-by-default.md

```diff
@@ -0,0 +1,79 @@
+- help improve the developer experience by encouraging its use
+
+## Detailed design
+
+- Add `ember-cli-fastboot` and `ember-cli-fastboot-testing` into the package.json blueprint and
+update the README.md in the blueprint to include a section on FastBoot.
+- Update `ember-cli-fastboot` to be *off* by default, and turn on if a query param or the `FASTBOOT_ENABLED` environment variable is set.
+variable exists.
+
+## How we teach this
+
+Because this RFC focuses on enabling *access* to SSR, rather than making SSR the default experience,
+the teaching material focuses on setting expectations for new applications, vs teaching about SSR
+as a technology for frontend frameworks.
+
+First and foremost, including FastBoot in the default application should set expectations that
+the developer experience when running FastBoot locally comes with a new set of challenges that
+Ember developers are not accustomed to. To be fair, these challenges should not be completely
+unfamiliar to those who work in full stack applications today (live reload, server errors, etc).
+
+Secondly, rendering Ember templates on the server comes with some caveats, such as lack of DOM access
+and browser APIs. These caveats are addressed partially by features such as Element Modifiers that
+no-op in FastBoot, and are documented on ember-fastboot.com, but they would need to be a core part
+of teaching Ember as well. The majority of these caveats relate to object initialization, and there
+may be a clever way of teaching this in one place.
+
+Third, Ember's current approach to testing would need to account for writing page rendering tests
+through the same server that is built into `ember serve`. This means creating generators for
+Node-style tests that make requests directly to the server, rather than using the `visit()`
+API to go to a URL. There would need to be a blueprint for this and possibly a new set of test helpers.
+
+Lastly, deploying an Ember Application with FastBoot will need to explain that Ember apps built
+with FastBoot turned on will not behave differently unless served by a Node server. In order to
+make this absolutely clear, the Deployment guide will need to cover this.
+
+## Drawbacks
+
+Adding FastBoot in every application adds complexity to application development
+(testing, deployment, etc). These can generally increase Ember's surface area and make it harder
+to learn. I believe this can be mitigated by leaving FastBoot and all its various
+pieces turned off by default, and making it clear that the built-in experience will take time to
+mature.
+
+## Alternatives
+
+Add Server Side rendering as a top level section to the Guides on emberjs.com. This could be an
+effective signal, but I don't think it goes far enough to accomplish our goal of improving the web.
+
+## Unresolved questions
+
+- Should `ember-cli-fastboot` be integrated with `fastboot-app-server` before it is integrated into Ember as a default?
```

Comment:
Even if it's not integrated with `ember-cli-fastboot`, would it be worth it to add `fastboot-app-server` to the default blueprint as well? I've switched my fastboot app to use it for some time now and it's been a great experience.